frontend/packages/ux-editor/src/containers/FormDesigner.tsx

```diff
@@ -40,74 +39,90 @@ export interface FormDesignerProps {
+    org,
+    app,
+    selectedLayoutSet
+  );
   const { data: formLayoutSettings } = useFormLayoutSettingsQuery(org, app, selectedLayoutSet);
   const { data: ruleModel } = useRuleModelQuery(org, app, selectedLayoutSet);
   const { isSuccess: isRuleConfigFetched } = useRuleConfigQuery(org, app, selectedLayoutSet);
   const addLayoutMutation = useAddLayoutMutation(org, app, selectedLayoutSet);
-  const layoutOrder = useMemo(() => formLayouts?.[selectedLayout]?.order || {}, [formLayouts, selectedLayout]);
+  const layoutOrder = useMemo(
+    () => formLayouts?.[selectedLayout]?.order || {},
+    [formLayouts, selectedLayout]
+  );
   const t = useText();
 
   const layoutPagesOrder = formLayoutSettings?.pages.order;
 
   const formLayoutIsReady =
-    instanceId &&
-    formLayouts &&
-    formLayoutSettings &&
-    ruleModel &&
-    isRuleConfigFetched;
+    instanceId && formLayouts && formLayoutSettings && ruleModel && isRuleConfigFetched;
 
   const mapErrorToDisplayError = (): { title: string; message: string } => {
     const defaultTitle = t('general.fetch_error_title');
     const defaultMessage = t('general.fetch_error_message');
 
     const createErrorMessage = (resource: string): { title: string; message: string } => ({
       title: `${defaultTitle} ${resource}`,
-      message: defaultMessage
+      message: defaultMessage,
     });
 
     if (layoutFetchedError) {
       return createErrorMessage(t('general.layout'));
     }
-  }
+  };
 
   /**
    * Set the correct selected layout based on url parameters
    */
   useEffect(() => {
-    
     const firstLayoutPage = layoutPagesOrder?.[0];
     if (!firstLayoutPage) return;
 
     const localStorageLayout = getSelectedLayoutInLocalStorage(instanceId);
     const searchParamsLayout = searchParams.get('layout');
 
-    const selectFirstLayoutPage = () => {
-      setSearchParams({ ...deepCopy(searchParams), layout: firstLayoutPage });
+    const updateLayoutInSearchParams = (layout: string) => {
+      setSearchParams((prevParams) => ({ ...prevParams, layout }));
     };
-    
+
     const isValidLayout = (layoutName: string): boolean => {
       const isExistingLayout = layoutPagesOrder?.includes(layoutName);
       const isReceipt = formLayoutSettings?.receiptLayoutName === layoutName;
-      return (isExistingLayout || isReceipt);
+      return isExistingLayout || isReceipt;
     };
 
-    if (isValidLayout(localStorageLayout)) {
-      dispatch(FormLayoutActions.updateSelectedLayout(localStorageLayout));
-      setSearchParams({ ...deepCopy(searchParams), layout: localStorageLayout });
-    } else if (isValidLayout(searchParamsLayout)) {
+    if (isValidLayout(searchParamsLayout)) {
       dispatch(FormLayoutActions.updateSelectedLayout(searchParamsLayout));
       setSelectedLayoutInLocalStorage(instanceId, searchParamsLayout);
-    } else {
-      selectFirstLayoutPage();
+      return;
+    }
+
+    if (isValidLayout(localStorageLayout)) {
```

Comment:
The fix for the feature flag issue was to swap places between the checks for localStorage and SearchParams